clients/src/main/java/org/apache/kafka/clients/admin/DescribeUserScramCredentialsResult.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.annotation.InterfaceStability;
+import org.apache.kafka.common.errors.ResourceNotFoundException;
+import org.apache.kafka.common.message.DescribeUserScramCredentialsResponseData;
+import org.apache.kafka.common.protocol.Errors;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+/**
+ * The result of the {@link Admin#describeUserScramCredentials()} call.
+ *
+ * The API of this class is evolving, see {@link Admin} for details.
+ */
+@InterfaceStability.Evolving
+public class DescribeUserScramCredentialsResult {
+    private final KafkaFuture<DescribeUserScramCredentialsResponseData> dataFuture;
+
+    /**
+     * Package-private constructor
+     *
+     * @param dataFuture the future indicating response data from the call
+     */
+    DescribeUserScramCredentialsResult(KafkaFuture<DescribeUserScramCredentialsResponseData> dataFuture) {
+        this.dataFuture = Objects.requireNonNull(dataFuture);
+    }
+
+    /**
+     *
+     * @return a future for the results of all described users with map keys (one per user) being consistent with the
+     * contents of the list returned by {@link #users()}. The future will complete successfully only if all such user
+     * descriptions complete successfully.
+     */
+    public KafkaFuture<Map<String, UserScramCredentialsDescription>> all() {
+        return KafkaFuture.allOf(dataFuture).thenApply(v -> {
```

Comment:
There's no reason to call `allOf` on a single future.  After all, `allOf`'s function is convert multiple futures to a single one.  But if you already have a single future, this is not needed.

You could use `dataFuture.thenApply`, but that will not trigger if `dataFuture` is completed exceptionally.

Instead, what you want here is something like the following:
`dataFuture.whenComplete( (data, exception) -> if (exception != null) { ... error ... }  else { ... handle data ... }`